monai/fl/client/monai_algo.py

```diff
@@ -121,11 +163,16 @@ def train(self, data: ExchangeObject, extra=None):
                 data = _filter(data, extra)
         self.phase = FlPhase.TRAIN
         self.logger.info(f"Load {self.client_name} weights...")
-        local_weights = convert_global_weights(
+        self.global_weights = convert_global_weights(
             global_weights=data.weights, local_var_dict=self.trainer.network.state_dict()
         )
 
-        self.trainer.network.load_state_dict(local_weights)
+        # set engine state max epochs.
+        self.trainer.state.max_epochs = self.trainer.state.epoch + self.local_epochs
+        # get current iteration when a round starts
+        self.iter_of_start_time = self.trainer.state.iteration
+
+        self.trainer.network.load_state_dict(self.global_weights)
```

Comment:
I would suggest to use the `copy_model_state` function of MONAI utils to enhance the logic here:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/utils.py#L396

Thanks.

Reply:
Thanks. Using it here now and at other places.